viewer.py

```diff
@@ -307,30 +313,49 @@ def check_update():
     """
 
     sha_file = path.join(settings.get_configdir(), 'latest_screenly_sha')
+    device_id_file = path.join(settings.get_configdir(), 'device_id')
 
     if path.isfile(sha_file):
         sha_file_mtime = path.getmtime(sha_file)
         last_update = datetime.fromtimestamp(sha_file_mtime)
     else:
         last_update = None
 
+    if not path.isfile(device_id_file):
+        device_id = ''.join(random.choice(string.ascii_lowercase + string.digits) for _ in range(15))
+        with open(device_id_file, 'w') as f:
+            f.write(device_id)
+    else:
+        with open(device_id_file, 'r') as f:
+            device_id = f.read()
+
     logging.debug('Last update: %s' % str(last_update))
 
     git_branch = sh.git('rev-parse', '--abbrev-ref', 'HEAD').strip()
+    git_hash = sh.git('rev-parse', '--short', 'HEAD').strip()
+
     if last_update is None or last_update < (datetime.now() - timedelta(days=1)):
 
-        if not url_fails('http://stats.screenlyapp.com'):
-            latest_sha = req_get('http://stats.screenlyapp.com/latest/{}'.format(git_branch))
+        if not settings['analytics_opt_out'] or not is_ci():
```

Comment:
it seems to me that this should be "and"

Reply:
No, because we don't want that code to run on CI ever as it would skew the data.